### PR TITLE
Feat: Swoole table update mid-request

### PR DESCRIPTION
**Pull request by Meldiron**
Prevents race condition when we forward to Random executors when actually a runtime is being prepared on one of them.

### CODE REVIEW

File: app/http.php [hunk 1]
```diff
@@ -199,8 +200,34 @@ function logError(Throwable $error, string $action, ?Logger $logger, Utopia\Rout
...
         }
 
+        /**
+         * @var string $hostname
+         */
         $hostname = $option->getState('hostname') ?? '';
 
+        // Optimistic update. Mark runtime up instantly to prevent race conditions
+        // Next health check with confirm it started well, and update usage stats
+        $runtimeId = $request->getHeader('x-opr-runtime-id', '');
```
Comment (eldadfux) [inline app/http.php:210]: Can we try to get the $runtimeId from the URL path?

File: app/http.php [hunk 2]
```diff
@@ -199,8 +200,34 @@ function logError(Throwable $error, string $action, ?Logger $logger, Utopia\Rout
...
+        if (!empty($runtimeId)) {
+            /**
+             * @var array<string,mixed> $stateItem
+             */
+            $stateItem = $state->get($hostname);
+
+            if ($stateItem['runtimes'] ?? null === null) {
```
Comment (eldadfux) [inline app/http.php:217]: I think there might be a better way of writing this. @christyjacob4 ?

File: app/http.php (hunk 1 above)
Comment (Meldiron) [reply app/http.php:251]: Keeping it as header so we can move quickly.

File: app/http.php [hunk 3]
```diff
@@ -230,18 +230,44 @@ function logError(Throwable $error, string $action, ?Logger $logger, Utopia\Rout
...
+        // Optimistic update. Mark runtime up instantly to prevent race conditions
+        // Next health check with confirm it started well, and update usage stats
+        $runtimeId = $request->getHeader('x-opr-runtime-id', '');
+        if (!empty($runtimeId)) {
+            /**
+             * @var array<string,mixed> $stateItem
+             */
+            $stateItem = \json_decode($state->get($hostname)['state'] ?? '{}', true);
+
+            if ($stateItem['runtimes'] === null) {
```
Comment (christyjacob4) [inline app/http.php:258]:
```suggestion
            if (!isset($stateItem['runtimes'])) {
```
You can bring the phpstan level back to max and use `isset` instead

File: app/http.php [hunk 4]
```diff
@@ -230,18 +230,44 @@ function logError(Throwable $error, string $action, ?Logger $logger, Utopia\Rout
...
+                $stateItem['runtimes'] = [];
+            }
+
+            if ($stateItem['runtimes'][$runtimeId] === null) {
```
Comment (christyjacob4) [inline app/http.php:262]: You can bring the phpstan level back to max and use `isset` instead ( like above )

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Last set of comments

File: app/http.php (hunk 3 above)
Comment (Meldiron) [reply app/http.php]: Used isset, PHPstan still complains. Level 8 is fine for now.

File: app/http.php (hunk 4 above)
Comment (Meldiron) [reply app/http.php]: Used isset, PHPstan still complains. Level 8 is fine for now.

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**